### PR TITLE
Upgrade to AWSSpring 3.1

### PR DESCRIPTION
**Pull request by danielrohe**
Upgrades of this library to AWSSpring 3.1 including restructuring and following the structure imposed by AWSSpring

### CODE REVIEW

File: zalando-cloud-aws-samples/zalando-cloud-aws-kms-sample/README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+[java.base/java.lang.Thread.getStackTrace(Thread.java:1610), 
+  com.zalando.awsspring.cloud.bootstrap.encrypt.KmsTextEncryptor.decrypt(KmsTextEncryptor.java:70), 
+  org.springframework.cloud.bootstrap.encrypt.AbstractEnvironmentDecrypt.decrypt(AbstractEnvironmentDecrypt.java:143), 
+  org.springframework.cloud.bootstrap.encrypt.AbstractEnvironmentDecrypt.lambda$decrypt$0(AbstractEnvironmentDecrypt.java:136),
+  java.base/java.util.LinkedHashMap.replaceAll(LinkedHashMap.java:731),
+  org.springframework.cloud.bootstrap.encrypt.AbstractEnvironmentDecrypt.decrypt(AbstractEnvironmentDecrypt.java:131), 
+  org.springframework.cloud.bootstrap.encrypt.AbstractEnvironmentDecrypt.decrypt(AbstractEnvironmentDecrypt.java:70), 
+  org.springframework.cloud.bootstrap.encrypt.EnvironmentDecryptApplicationInitializer.initialize(EnvironmentDecryptApplicationInitializer.java:95),
```
Comment (oussewa) [inline zalando-cloud-aws-samples/zalando-cloud-aws-kms-sample/README.md:35]: What is this?

Comment (danielrohe) [reply zalando-cloud-aws-samples/zalando-cloud-aws-kms-sample/README.md:35]: this is a test that shows the current AWSSpring is calling decrypt twice

**Comment by danielrohe** [comment]
:+1:

**Comment by oussewa** [comment]
👍

**Pull request merged**